### PR TITLE
netatalk: update to 3.1.18

### PR DESCRIPTION
**Pull request by APCCV**
Maintainer: @APCCV / Antonio Pastor

Commit restores package after it was removed from OpenWrt 21.02.

Compile tested: ipq806x - OpenWrt 23.05.3, r23809
- Compiled using SDK for 23.05.3.

Run tested: ipq806x - OpenWrt 23.05.3, r23809. Tests included:
- opkg install of package
- edit config file to create network shares
- connect to shares from MacOS (both vintage 10.13 and 14.4)
- stored, modify and delete files in share from Finder
- check and update file Information (properties) from Finder
- configured share to be used for TimeMachine backup
- create a backup (full backup - initial backup)
- create incremental backups
- enter Time Machine and browse backups

Description: 
Netatalk package was removed in 2023 as it was stale and used old 3.1.13 version from 2018.
This commit uses the latest stable netatalk version 3.1.18.
A separate PR will be created to take the package out of packages-abandoned.

### CODE REVIEW

File: net/netatalk/Makefile [hunk 1]
```diff
@@ -0,0 +1,99 @@
+#
+# Copyright (C) 2009-2013 OpenWrt.org
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
```
Comment (PolynomialDivision) [inline net/netatalk/Makefile:6]: please use spdx identifier

**Comment by APCCV** [comment]
Rats. Clearly I don't know (yet) how to do this. Squashing commits is an art I don't manage yet.
Apologies for the mess... will open a new PR.

File: net/netatalk/Makefile (hunk 1 above)
Comment (APCCV) [reply net/netatalk/Makefile:6]: Well, made a mess out of it, but done.

**Comment by APCCV** [comment]
Created https://github.com/openwrt/packages/pull/24257

**Pull request closed without merging**